test/integration/component/test_secsr_mount.py

```diff
@@ -0,0 +1,170 @@
+            cloud.template_spool_ref.template_id=cloud.vm_template.id and cloud.vm_template.uuid='%s';" %
+            id)
+        self.logger.debug("Qresult %s" % format(qresultset))
+        if not isinstance(qresultset, list) or len(qresultset) <= 1:
+            return False, 1
+        
+        for result in qresultset:
+            self.logger.debug('Template %s download status %s' % (result[0], result[1])) 
+            if result[1] != 'DOWNLOADED':
+                return False, 1
+            
+        return True, 1
+
+    @attr(
+        tags=[
+            "advanced",
+            "xenserver"],
+        required_hardware="true")
+    def test_01_prepare_template_local_storage(self):
+    
+        if not(self.isOnlyLocalStorageAvailable()):
+            raise unittest.SkipTest("Skipping this test as this is for Local storage on only.");
+            return
+        
+        listHost = Host.list(
+            self.apiclient,
+            type='Routing',
+            zoneid=self.zone.id,
+            podid=self.pod.id,
+        )
+        for host in listHost:
+            self.logger.debug('Host id %s, hypervisor %s, localstorage %s' % (host.id, host.hypervisor, host.islocalstorageactive))
+                  
+        if len(listHost) < 2:
+            self.logger.debug("Prepare secondary storage race condition can be tested with two or more host only %s, found" % len(listHost));
+            raise unittest.SkipTest("Prepare secondary storage can be tested with two host only %s, found" % len(listHost));
+            return
+        
+        list_template_response = Template.list(
+                                            self.apiclient,
+                                            templatefilter='all',
+                                            zoneid=self.zone.id)
+        
+        template_response = list_template_response[0]
+        
+        self.logger.debug('Template id %s is Ready %s' % (template_response.id, template_response.isready))
+        
+        if template_response.isready != True:
+            raise unittest.SkipTest('Template id %s is Not Ready' % (template_response.id));
+            
+        
+        try:
+            cmd = prepareTemplate.prepareTemplateCmd()
+            cmd.zoneid = self.zone.id
+            cmd.templateid = template_response.id
+            result = self.apiclient.prepareTemplate(cmd)
+            self.logger.debug('Prepare Template result %s' % result)
+        except Exception as e:
+           raise Exception("Warning: Exception during prepare template : %s" % e)
+    
+        downloadStatus = wait_until(self.services['sleep'], self.services['timeout'], self.areDownloaded, template_response.id)
+        if not(downloadStatus): 
+            raise Exception("Prepare template failed to download template to primary stores or is taking longer than expected")
+                  
+        return
```

Comment:
Remove blank `return`, we've already reached the end of the method.

Reply:
fixed